File: NewVOCAPlus/VOCAPlus/IntUtly.vb
Imports System.Globalization
Imports System.Text.RegularExpressions

Public NotInheritable Class IntUtly

    Public Shared Sub ValdtInt(ByVal e As KeyPressEventArgs) ' numeric only integer
        If Not Char.IsControl(e.KeyChar) AndAlso (Not Char.IsDigit(e.KeyChar)) Then
            e.Handled = True
        End If
    End Sub
    ' ***************************************
    Public Shared Sub ValdtNumber(sender As Object, e As KeyPressEventArgs) ' numeric decmal
        Dim TempNum As TextBox = sender
        If (Not e.KeyChar = ChrW(Keys.Back) And ("0123456789.").IndexOf(e.KeyChar) = -1) Or (e.KeyChar = "." And TempNum.Text.ToCharArray().Count(Function(c) c = ".") > 0) Then
            e.Handled = True
        End If
    End Sub
    Public Shared Sub ValdtIntLetter(ByVal e As KeyPressEventArgs) ' numeric & Letters & White Space & Backspace
        If Not Char.IsControl(e.KeyChar) AndAlso (Char.IsDigit(e.KeyChar)) OrElse (Char.IsLetter(e.KeyChar)) OrElse (Char.IsWhiteSpace(e.KeyChar)) OrElse (e.KeyChar = ChrW(Keys.Back)) OrElse (e.KeyChar = ChrW(Keys.ShiftKey)) AndAlso (Char.IsLetter(e.KeyChar)) Then
        Else
            e.Handled = True
        End If
    End Sub
    Public Shared Sub ValdtLetter(ByVal e As KeyPressEventArgs) ' numeric & Letters & White Space & Backspace
        If Not Char.IsControl(e.KeyChar) AndAlso (Char.IsLetter(e.KeyChar)) OrElse (Char.IsWhiteSpace(e.KeyChar)) OrElse (e.KeyChar = ChrW(Keys.Back)) OrElse (e.KeyChar = ChrW(Keys.ShiftKey)) AndAlso (Char.IsLetter(e.KeyChar)) Then
        Else
            e.Handled = True
        End If
    End Sub
    Public Shared Sub ValdtNotNull(ByVal sender As Object, ByVal e As System.EventArgs)
        Dim No As TextBox = sender
        If No.Text.Trim = "" Then
            MsgBox("This field Must be filled!")
            No.Focus()
        End If
    End Sub
    '********************************************************
    Public Shared Function IsValidEmail(email As String) As Boolean
        If String.IsNullOrWhiteSpace(email) Then Return False
        Try ' Use IdnMapping class to convert Unicode domain names.
            ' Examines the domain part of the email and normalizes it.
            Dim DomainMapper =
                Function(match As Match) As String
                    Dim idn = New IdnMapping '                                      Use IdnMapping class to convert Unicode domain names.
                    Dim domainName As String = idn.GetAscii(match.Groups(2).Value) 'Pull out and process domain name (throws ArgumentException on invalid)
                    Return match.Groups(1).Value & domainName
                End Function
            email = Regex.Replace(email, "(@)(.+)$", DomainMapper, RegexOptions.None, TimeSpan.FromMilliseconds(200)) 'Normalize the domain
        Catch e As RegexMatchTimeoutException
            Return False
        Catch e As ArgumentException
            Return False

        End Try
        Try
            Return Regex.IsMatch(email,
                                 "^(?("")("".+?(?<!\\)""@)|(([0-9a-z]((\.(?!\.))|[-!#\$%&'\*\+/=\?\^`\{\}\|~\w])*)(?<=[0-9a-z])@))" +
                                 "(?(\[)(\[(\d{1,3}\.){3}\d{1,3}\])|(([0-9a-z][-0-9a-z]*[0-9a-z]*\.)+[a-z0-9][\-a-z0-9]{0,22}[a-z0-9]))$",
                                 RegexOptions.IgnoreCase, TimeSpan.FromMilliseconds(250))
        Catch e As RegexMatchTimeoutException
            Return False
        End Try
    End Function
End Class

File: CC/VOCAC/VOCAC/BL/IntUtly.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Globalization;
using System.Text.RegularExpressions;
using System.Media;

namespace VOCAC.BL
{
    public static class IntUtly
    {
        public static void ValdtInt(KeyPressEventArgs e) // numeric only int
        {
            if (Char.IsControl(e.KeyChar) == false && Char.IsDigit(e.KeyChar) == false)
            {
                e.Handled = true;
                SystemSounds.Beep.Play();
            }
        }
        public static void ValdtAll(KeyPressEventArgs e) // numeric only int
        {
            if ((e.KeyChar)==8)
            {

            }
            else if (char.IsControl(e.KeyChar) == true && (Keys)e.KeyChar != Keys.V)

            {
                e.Handled = true;
                SystemSounds.Beep.Play();
            }
        }
        public static void ValdtNumber(object sender, KeyPressEventArgs e) //numeric decmal
        {
            TextBox TempNum = (TextBox)sender;
            if (((Keys)e.KeyChar != Keys.Back && ("0123456789.").IndexOf(e.KeyChar) == -1) ||
                e.KeyChar == Convert.ToChar(".") && TempNum.Text.ToCharArray().Count(c => c == Convert.ToChar(".")) > 0)
            {
                e.Handled = true;
                SystemSounds.Beep.Play();
            }
        }
        public static void ValdtIntLetter(KeyPressEventArgs e) // numeric & Letters & White Space & Backspace
        {
            if (char.IsControl(e.KeyChar) == true && (char.IsDigit(e.KeyChar)) ||
                                             (char.IsLetter(e.KeyChar)) ||
                                             (char.IsWhiteSpace(e.KeyChar)) ||
                                               (Keys)e.KeyChar == Keys.Back ||
                ((Keys)e.KeyChar == Keys.ShiftKey) && (char.IsLetter(e.KeyChar)))
            { }
            else
            {
                e.Handled = true;
                SystemSounds.Beep.Play();
            }
        }
        public static void ValdtLetter(object sender, KeyPressEventArgs e) //  Letters & White Space & Backspace
        {
            TextBox TempNum = (TextBox)sender;
            if (char.IsControl(e.KeyChar) == false && (char.IsLetter(e.KeyChar)) ||
                                                      (char.IsWhiteSpace(e.KeyChar)) ||
                                                        (Keys)e.KeyChar == Keys.Back ||
                ((Keys)e.KeyChar == Keys.ShiftKey) && char.IsLetter(e.KeyChar))
            { }
            else
            {
                e.Handled = true;
                SystemSounds.Beep.Play();
            }
        }
        public static void ValdtNotNull(object sender, EventArgs e) //  Check Null Value
        {
            TextBox No = (TextBox)sender;
            if (No.Text.Trim() == "")
            {
                SystemSounds.Beep.Play();
                No.Focus();
            }
        }
    }
}
